lib/wasix/src/runtime/resolver/backend_source.rs

```diff
@@ -217,7 +217,7 @@ impl BackendSource {
     async fn query_by_hash(
         &self,
         hash: &PackageHash,
-    ) -> Result<Option<PackageSummary>, QueryError> {
+    ) -> Result<Option<PackageSummary>, anyhow::Error> {
```

Comment:
Are we returning any other error than `QueryError`, for this to be needed?

Reply:
This is a private helper function that never returns a specific QueryError, so I changed the return type to avoid having to  write `.map_err()` 5 times.